### PR TITLE
introduce support for RPC v0.2.1

### PR DESCRIPTION
**Pull request by gregoryguillou**
see #174 . This PR:
- adds a v0.2.1 RPC implementation. Noone is using it
- does some cleaning of v0.1.0 RPC and remove the contracts from the directory
- adds a Github Action with devnet 0.3.5 to test the new implementation
- changes the `version` format from an uint64 to a *big.Int to support format used to run `EstimateFee`

### CODE REVIEW

File: rpcv02/contract.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	if err := do(ctx, provider.c, "starknet_getStorageAt", &value, contractAddress, hashKey, blockID); err != nil {
+		return "", err
+	}
+	return value, nil
+}
+
+// Nonce returns the Nonce of a contract
+func (provider *Provider) Nonce(ctx context.Context, contractAddress types.Hash) (*string, error) {
+	nonce := ""
+	if err := do(ctx, provider.c, "starknet_getNonce", &nonce, contractAddress); err != nil {
```
Comment (archseer) [inline rpcv02/contract.go]: Nonce needs a `block_id` parameter before contract address

File: rpcv02/contract.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	"context"
+	"fmt"
+
+	"github.com/dontpanicdao/caigo/types"
+)
+
+// Class gets the contract class definition associated with the given hash.
+func (provider *Provider) Class(ctx context.Context, classHash string) (*types.ContractClass, error) {
+	var rawClass types.ContractClass
+	if err := do(ctx, provider.c, "starknet_getClass", &rawClass, classHash); err != nil {
```
Comment (archseer) [inline rpcv02/contract.go]: Similar change here, needs to take a blockID before `classHash`

**Comment by archseer** [comment]
Sorry if I'm reviewing this too soon, I can take another look when this draft is ready!

**Comment by gregoryguillou** [comment]
> Sorry if I'm reviewing this too soon, I can take another look when this draft is ready!

It is too soon ;-). Do not be sorry though, I am very happy you help! so please continue. Maybe just wait I update the PR to ready

**Comment by archseer** [comment]
I'll test this out against https://github.com/smartcontractkit/chainlink-starknet/pull/172 tomorrow!

**Comment by archseer** [comment]
Hmm `account` is still missing a `ProviderRPCv02`/`NewRPCAccount` that would accept v0.2

**Comment by gregoryguillou** [comment]
> Hmm `account` is still missing a `ProviderRPCv02`/`NewRPCAccount` that would accept v0.2

yes. I will work on this one...

**Comment by archseer** [comment]
Seems to work well for us 👍🏻

**Comment by gregoryguillou** [comment]
This PR is still missing some verbs to fully support v0.2.1. I will merge it, however and work on top of the `main` branch.

**Pull request merged**